dragonfly/block/kelp.go

```diff
@@ -0,0 +1,128 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/dragonfly/entity/physics"
+	"github.com/df-mc/dragonfly/dragonfly/item"
+	"github.com/df-mc/dragonfly/dragonfly/world"
+	"github.com/go-gl/mathgl/mgl64"
+	"math/rand"
+	"time"
+)
+
+type Kelp struct {
+
+	//0-15, defines the age of the kelp block. If age is 15, kelp won't grow any further.
+	Age int
+}
+
+//Kelp can be instantly destroyed.
+func (k Kelp) BreakInfo() BreakInfo {
+	return BreakInfo{
+		Hardness:    0.0,
+		Harvestable: alwaysHarvestable,
+		Effective:   nothingEffective,
+		Drops:       simpleDrops(item.NewStack(Kelp{}, 1)),
+	}
+}
+
+func (Kelp) EncodeItem() (id int32, meta int16) {
+	return 335, 0
+}
+
+func (k Kelp) EncodeBlock() (name string, properties map[string]interface{}) {
+	return "minecraft:kelp", map[string]interface{}{"age": int32(k.Age)}
+}
+
+func (Kelp) LightLevelRequired() uint8 {
+	return 0 //Kelp grows without any light.
+}
+
+func (Kelp) RequiresHydration() bool {
+	return false //Kelp does not need hydration but rather just has to be placed in water.
+}
+
+func (Kelp) RequiresFarmland() bool {
+	return false //Kelp doesn't require a farmland, literally grows underwater.
+}
+
+func (Kelp) CanDisplace(b world.Liquid) bool {
+	_, water := b.(Water)
+	return water //Kelp can waterlog.
+}
+
+func (Kelp) SideClosed(pos, side world.BlockPos, w *world.World) bool {
+	return false //Kelp can always be flowed through.
+}
+
+func (Kelp) AABB(world.BlockPos, *world.World) []physics.AABB {
+	return nil //Kelp can be placed even if someone is standing on its placement position.
+}
+
+//In Java Edition, the age value can be up to 25, but MCPE limits it to 15.
+func (k Kelp) SetRandomAge() Kelp {
```

Comment:
Should make this an unexported method as it's quite useless outside of this package.